src/order/chain.lean

```diff
@@ -277,14 +282,22 @@ protected lemma max_chain (s : flag α) : is_max_chain (≤) (s : set α) := ⟨
 lemma top_mem [order_top α] (s : flag α) : (⊤ : α) ∈ s := s.max_chain.top_mem
 lemma bot_mem [order_bot α] (s : flag α) : (⊥ : α) ∈ s := s.max_chain.bot_mem
 
+/-- Reinterpret a maximal chain as a flag. -/
+@[simps] protected def _root_.is_max_chain.flag (hc : is_max_chain (≤) c) : flag α :=
+⟨c, hc.is_chain, hc.2⟩
```

Comment:
Does `.2` have a more revealing name, or is it just `right`?

Reply:
No. `is_max_chain` is just a wrapper around `and`, and there's no specialised projection.